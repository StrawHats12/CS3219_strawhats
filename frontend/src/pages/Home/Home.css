.hero {
  height: 450px;
  background-color: rgba(33, 37, 41, 1);
}

.hero-img {
  height: 250px;
}

.pr {
  padding-right: 1.5rem;
}

.pl-3 {
  padding-left: 3rem;
}

.heading {
  margin-top: 20px;
  text-align: center;
}
