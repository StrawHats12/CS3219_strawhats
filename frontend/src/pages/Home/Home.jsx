import { Container } from "react-bootstrap";
import "./Home.css";
import logo from "./logo192.png";
import hero from "./hero.png";
import { useEffect, useState } from "react";
import { getAllListings } from "../../services/listings-service";
import StrawhatSpinner from "../../components/StrawhatSpinner";
import ListingsCardVertical from "../../components/Listings/ListingsCardVertical";

const Home = () => {
  const [listings, setListings] = useState(null);
  const [isLoading, setIsLoading] = useState(true);

  useEffect(() => {
    getAllListings().then((res) => {
      if (!res) {
        setIsLoading(false);
        return;
      }
      
      setListings(res.slice(0,9));
      setIsLoading(false);
    });
  }, []);

  return (
    <div>
      <div className="d-flex justify-content-around align-items-center text-white hero">
        <div>
          <div className="d-flex justify-content-left align-items-center">
            <img src={logo} alt="" style={{ width: "80px" }}></img>
            <div className="mx-2">
              <h1>Strawhats</h1>
              <h3>Auctions made easy.</h3>
            </div>
          </div>
          <ul className="pl-7 mt-3">
            <li>
              <h5>Live Video Streaming</h5>
            </li>
            <li>
              <h5>Easy to use</h5>
            </li>
            <li>
              <h5>Free forever</h5>
            </li>
          </ul>
        </div>
        <img src={hero} alt="" className="hero-img" />
      </div>
      <Container>
        <h1 className="heading">Featured Listings</h1>
        <div className="d-flex justify-content-around align-items-center flex-wrap">
          {isLoading ? (
            <StrawhatSpinner />
          ) : (
            listings?.map((listing, i) => (
              <ListingsCardVertical listing={listing} key={i} />
            ))
          )}
        </div>
      </Container>
    </div>
  );
};

export default Home;
