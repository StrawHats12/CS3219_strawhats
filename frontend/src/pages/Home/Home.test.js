import renderer from "react-test-renderer";
import Home from ".";
const axios = require("axios");

jest.mock("axios");

test("Home page renders correctly", async () => {
  axios.get.mockResolvedValue([]);

  const tree = renderer.create(<Home />).toJSON();
  expect(tree).toMatchSnapshot();
});
