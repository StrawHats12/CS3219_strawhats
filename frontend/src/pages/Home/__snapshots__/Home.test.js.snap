// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page renders correctly 1`] = `
<div>
  <div
    className="d-flex justify-content-around align-items-center text-white hero"
  >
    <div>
      <div
        className="d-flex justify-content-left align-items-center"
      >
        <img
          alt=""
          src="logo192.png"
          style={
            Object {
              "width": "80px",
            }
          }
        />
        <div
          className="mx-2"
        >
          <h1>
            Strawhats
          </h1>
          <h3>
            Auctions made easy.
          </h3>
        </div>
      </div>
      <ul
        className="pl-7 mt-3"
      >
        <li>
          <h5>
            Live Video Streaming
          </h5>
        </li>
        <li>
          <h5>
            Easy to use
          </h5>
        </li>
        <li>
          <h5>
            Free forever
          </h5>
        </li>
      </ul>
    </div>
    <img
      alt=""
      className="hero-img"
      src="hero.png"
    />
  </div>
  <div
    className="container"
  >
    <h1
      className="heading"
    >
      Featured Listings
    </h1>
    <div
      className="d-flex justify-content-around align-items-center flex-wrap"
    >
      <div
        className="mx-auto text-center container"
      >
        <div
          className="spinner-border"
          role="status"
        >
          <span
            className="visually-hidden"
          >
            Loading...
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
